docs/dev/lsp-extensions.md

```diff
@@ -385,6 +385,17 @@ rust-analyzer supports only one `kind`, `"cargo"`. The `args` for `"cargo"` look
 }
 ```
 
+The args for `"shell"` look like this:
+
+```typescript
+{
+    kind: string;
```

Comment:
How should the LSP client use the `kind`  field?  Does it describe what the runnable does? Then how does it differ then from `label` field of the parent `Runnable`?

Also should the "shell" kind have a `expectTest` field just like the "cargo" kind?

BTW, how should the LSP client interpret the `workspaceRoot`, `expectTest`, and `overrideCargo` fields?

Additionally, I think it would make sense to copy the example from the Pull Request into the lsp-extension.md file, or just reference a user-level documentation describing this feature.

Thanks

Reply:
Also, shouldn't this PR explain the new build_info field of rust-project.json in manual.adoc?

Reply:
When `expectTest` is set the client should set the `UPDATE_EXPECT` env var to `1` when invoking the runnable. ideally there should be a field for arbitrary env var overrides in the payload, then this field could be replaced by the server instructing the client to just set the env var with that general mechanism as we'll need this for other things as well.

`overrideCargo` I think should just replace the cargo command that is being used by the client.